client/my-sites/themes/single-site-wpcom.jsx

```diff
@@ -14,33 +18,26 @@ import { getActiveTheme } from 'calypso/state/themes/selectors';
 import { connectOptions } from './theme-options';
 import ThemeShowcase from './theme-showcase';
 
-const ConnectedSingleSiteWpcom = connectOptions( ( props ) => {
-	const { currentPlan, currentThemeId, isVip, requestingSitePlans, siteId, siteSlug, translate } =
-		props;
-
-	const displayUpsellBanner = ! requestingSitePlans && currentPlan && ! isVip && siteId;
-	const upsellUrl = `/plans/${ siteSlug }`;
-	let upsellBanner = null;
-	if ( displayUpsellBanner ) {
-		const commonProps = {
-			className: 'themes__showcase-banner',
-			event: 'calypso_themes_list_install_themes',
-			feature: FEATURE_UPLOAD_THEMES,
-			plan: PLAN_BUSINESS,
-			title: translate( 'Upload your own themes with our Business and eCommerce plans!' ),
-			callToAction: translate( 'Upgrade now' ),
-			showIcon: true,
-		};
-
-		if ( isEnabled( 'themes/premium' ) ) {
-			if ( currentPlan.productSlug === PLAN_PREMIUM ) {
-				upsellBanner = <UpsellNudge { ...commonProps } />;
-			}
-		} else {
-			upsellBanner = <UpsellNudge { ...commonProps } />;
-		}
+const getUpgradeBannerForPlan = ( planSlug ) => {
+	if ( WPCOM_PREMIUM_PLANS.includes( planSlug ) ) {
```

Comment:
What if we display the banner on free/personal too? 

Can we use the `siteHasFeature` selector instead of hardcoding the plans? 🤔 Something like 

` if (! siteHasFeature( state, siteId, FEATURE_UPLOAD_THEMES ) ) { display banner`

Reply:
That's not the main idea. The original issue before this one was about removing the banner on personal plans, and free plans didn't show this banner at all. 

There are many other better upsells than this one since it's shown very deep down on the TS.